CS410_Assignment2-master/src/SensorPanel.java

```diff
@@ -0,0 +1,59 @@
+import java.awt.Color;
+
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JProgressBar;
+import javax.swing.border.TitledBorder;
+
+import adaptors.AppSensor;
+
+public class SensorPanel extends JPanel{
+	private AppSensor sensor;
+	private double danger;
+	private double critical;
+	
+	
+	public SensorPanel(AppSensor sensor, double danger, double critical) {
+		this.sensor = sensor;
+		this.danger = danger;
+		this.critical = critical;
+		
+		setLayout(null); //allows for items to be moved in panel
+		
+		double value = sensor.getValue();
+		Color color;
+		String message = sensor.getReport();
+		//set the error messages and sensor values based on sensor output
```

Comment:
I believe this logic for message to color translation should be inside the Adapter as another function. It would be easily broken if another adapter had a different set of messages or colors associated with it.